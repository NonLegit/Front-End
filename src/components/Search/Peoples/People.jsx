import { Box, Typography } from '@mui/material';
import Avatar from '@mui/material/Avatar';
import { useEffect, useState } from 'react';
import numFormatter from '../../../utils/MembersNum';
import followRequest from './FollowServer';
import {
  TextP, Text, TextContainer, OneSuggeest, Joined,
} from './style';
/**
 * Search by people
 * @component
 * @property  {function} handleJoin send follow to backend
 * @property  {function} handleMouseIn when hover set buttom value to unfollow
 * @property  {function} handleMouseOut when out of hover set button value to follow

 * @return {React.Component} - Search by people
 */
function Peoples(props) {
  const { people } = props;
  const [follow, setFollow] = useState(true);
  const handleJoin = async () => {
    await setFollow((prev) => !prev);
    followRequest(people?.userName, !follow);
  };

  const [hover, setHover] = useState(false);

  const handleMouseIn = () => {
    setHover(true);
  };
  const handleMouseOut = () => {
    setHover(false);
  };
  useEffect(() => {
    setFollow(people?.isFollowed);
  }, []);
  return (
    <OneSuggeest sx={{ display: 'flex', justifyContent: 'space-between' }}>
      <Box sx={{ display: 'flex' }}>

        <Avatar alt="Remy Sharp" src={people?.profilePicture} />
        <Box sx={{ display: 'flex', flexDirection: 'column' }}>
          <TextContainer>
            <Text to={`/user/${people?.userName}`}>
              U/
              {people?.userName}
            </Text>
            <Typography component="span" mx="4px" sx={{ fontSize: '6px', display: 'flex', alignItems: 'center' }}>
              â€¢
            </Typography>
            <TextP>
              {numFormatter((people.postKarma * 1) + (people.commentKarma * 1))}
              {' '}
              karma
            </TextP>
          </TextContainer>
          <Typography component="span" mx="4px" sx={{ fontSize: '12px', display: 'flex', alignItems: 'center' }}>
            {people?.description}
          </Typography>

        </Box>
      </Box>
      <Joined
        variant="outlined"
        onClick={handleJoin}
        onMouseEnter={handleMouseIn}
        onMouseLeave={handleMouseOut}
      >
        {(follow) ? ((hover) ? 'Unfollow' : 'Following') : 'follow'}
      </Joined>
    </OneSuggeest>
  );
}
export default Peoples;
