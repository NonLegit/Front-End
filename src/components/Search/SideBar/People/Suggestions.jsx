import Avatar from '@mui/material/Avatar';
import { useState } from 'react';
import followRequest from '../../Peoples/FollowServer';
import numFormatter from '../../../../utils/MembersNum';
import {
  TextP, Text, TextContainer, OneSuggeest, Joined,
} from './style';
/**
 * people in sidebar entity
 * @component
 * @property  {function} handleJoin send follow to backend
 * @property  {function} handleMouseIn when hover set buttom value to unfollow
 * @property  {function} handleMouseOut when out of hover set button value to follow

 * @return {React.Component} - people in sidebar entity
 */
function Suggestions(props) {
  const { people } = props;
  const [follow, setFollow] = useState(people?.isFollowed);
  const handleJoin = () => {
    followRequest(people?.userName, follow);
    setFollow((prev) => !prev);
  };
  const [hover, setHover] = useState(false);

  const handleMouseIn = () => {
    setHover(true);
  };
  const handleMouseOut = () => {
    setHover(false);
  };
  return (
    <OneSuggeest>
      <Avatar alt="Remy Sharp" src={people?.profilePicture} />
      <TextContainer>
        <Text to={`/user/${people?.userName}`}>
          U/
          {people?.displayName}
        </Text>
        <TextP>
          {numFormatter(people.postKarma + people.commentKarma)}
          {' '}
          karma
        </TextP>
      </TextContainer>
      <Joined
        variant="outlined"
        onClick={handleJoin}
        onMouseEnter={handleMouseIn}
        onMouseLeave={handleMouseOut}
      >
        {(follow ? (hover ? 'Unfollow' : 'Following') : 'follow')}
      </Joined>
    </OneSuggeest>
  );
}
export default Suggestions;
