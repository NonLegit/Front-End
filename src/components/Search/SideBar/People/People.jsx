import { Box } from '@mui/material';
import Suggestions from './Suggestions';
import {
  PeopleString, ShowMore, Suggestion,
} from './style';
/**
 * Search by posts sidebar people section
 * @component

 * @return {React.Component} - Search by posts sidebar people section
 */
function People(props) {
  const { peoples } = props;
  return (
    <Box sx={{ width: 310 }}>
      <PeopleString>People</PeopleString>
      <Suggestion>
        {peoples?.map((people) => (

          <Suggestions people={people} />
        ))}
      </Suggestion>
      <Box sx={{ display: 'flex', justifyContent: 'cenetr' }}>
        <ShowMore>See More Communities</ShowMore>
      </Box>

    </Box>
  );
}
export default People;
