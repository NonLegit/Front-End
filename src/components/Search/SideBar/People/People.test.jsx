import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import People from './People';

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(<Router><People /></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
