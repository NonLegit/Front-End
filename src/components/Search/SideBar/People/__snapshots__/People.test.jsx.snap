// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-15o7b25"
>
  <h4
    className="css-15t0qw2"
  >
    People
  </h4>
  <div
    className="css-1e7j8ur"
  />
  <div
    className="MuiBox-root css-fikvv"
  >
    <a
      className="css-1m7wn3z"
      href="/"
      onClick={[Function]}
    >
      See More Communities
    </a>
  </div>
</div>
`;
